verta_build_and_publish.sh

```diff
@@ -3,18 +3,31 @@
 set -eo pipefail
 
 export JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF8
-export VERSION_SUFFIX="$(git rev-parse --abbrev-ref HEAD)"
-export VERSION_SUFFIX=$(echo $VERSION_SUFFIX | sed 's,/,-,g' | tr '[:upper:]' '[:lower:]')
-export PROJECT_VERSION=$(mvn help:evaluate -Dexpression=revision -q -DforceStdout)
-export PROJECT_VERSION=$(echo $PROJECT_VERSION | sed "s,-SNAPSHOT,-${VERSION_SUFFIX}-SNAPSHOT,g")
+export BRANCH_NAME="$(git rev-parse --abbrev-ref HEAD)"
+export VERSION_SUFFIX=$(echo $BRANCH_NAME | sed 's,/,-,g' | tr '[:upper:]' '[:lower:]')
+export PROJECT_REVISION=$(mvn help:evaluate -Dexpression=revision -q -DforceStdout)
+
+# require the revision to end in -SNAPSHOT
+if [ "$PROJECT_REVISION" == "${PROJECT_REVISION/%-SNAPSHOT/}" ]; then
+    echo The "revision" property in pom.xml must end with "-SNAPSHOT" for this script to work correctly.
+    echo Actual value: revision = $PROJECT_REVISION
+    exit 1
+fi
+
+# Insert branch name in project version
+export PROJECT_VERSION=${PROJECT_REVISION/%-SNAPSHOT/-${VERSION_SUFFIX}-SNAPSHOT}
+
+# When building verta main replace -SNAPSHOT with commit info
+if [ "$BRANCH_NAME" == "ln/deploy-main" ]; then
```

Comment:
Old?

Reply:
Nope. New for testing. :) 
Reverted already.